### PR TITLE
ROC-4566:CourtDetermination-PR2

### PR DESCRIPTION
**Pull request by varma02**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-4566

### Change description
Using courtDetermination model in all places in claimant response journey

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/main/app/claims/converters/claimantResponseConverter.ts [hunk 1]
```diff
@@ -32,8 +20,8 @@ export class ClaimantResponseConverter {
       if (draftClaimantResponse.freeMediation) {
         reject.freeMediation = draftClaimantResponse.freeMediation.option === FreeMediationOption.YES
       }
-      if (draftClaimantResponse.rejectionReason) {
-        reject.reason = draftClaimantResponse.rejectionReason.text
+      if (draftClaimantResponse.courtDetermination.rejectionReason) {
```
[the remark this answers is not in the document]
Comment (varma02) [reply src/main/app/claims/converters/claimantResponseConverter.ts]: this check is for `ResponseRejection`, so check is right...i will explain to you when we speak

[the remark this answers is not in the document]
Comment (andrewwa-kainos) [reply src/main/app/claims/converters/claimantResponseConverter.ts]: Is the `courtDetermination` guaranteed to be present or is there an NPE risk?

File: src/main/features/claimant-response/routes/payment-date.ts [hunk 2]
```diff
@@ -51,7 +52,7 @@ export class PaymentDatePage extends AbstractPaymentDatePage<DraftClaimantRespon
         courtOfferedPaymentIntention.paymentOption = PaymentOption.INSTALMENTS
         courtOfferedPaymentIntention.repaymentPlan = {
           firstPaymentDate: paymentPlanConvertedToDefendantFrequency.startDate,
-          instalmentAmount: paymentPlanConvertedToDefendantFrequency.instalmentAmount,
+          instalmentAmount: Math.round(paymentPlanConvertedToDefendantFrequency.instalmentAmount * 100) / 100,
```
Comment (andrewwa-kainos) [inline src/main/features/claimant-response/routes/payment-date.ts]: Do we have official guidance on what the rounding strategy should be? Javascript rounding weirdness can make this drop a penny. Some people recommend `num.toFixed(2)` (or `+num.toFixed(2)` to force it back to a number)

Comment (varma02) [reply src/main/features/claimant-response/routes/payment-date.ts]: i have seen Math.round used in other places so i used it not sure what rounding strategy is...i guess i am going to leave it like that for now

File: src/test/app/claims/converters/claimantResponseConverter.ts [hunk 3]
```diff
@@ -140,157 +95,152 @@ describe('claimant response converter ', () => {
...
     it(' Accept defendant offer with CCJ', () => {
       const draftClaimantResponse = createDraftClaimantResponseBaseForAcceptance(null,YesNoOption.YES)
       draftClaimantResponse.formaliseRepaymentPlan = new FormaliseRepaymentPlan(FormaliseRepaymentPlanOption.REQUEST_COUNTY_COURT_JUDGEMENT)
-      draftClaimantResponse.decisionType = DecisionType.DEFENDANT
+      draftClaimantResponse.courtDetermination = courtDetermination
       expect(converter.convertToClaimantResponse(draftClaimantResponse)).to.deep.eq({
         'type': 'ACCEPTATION',
         'amountPaid': 10,
-        'formaliseOption': 'CCJ'
+        'formaliseOption': 'CCJ',
```
Comment (srineelesh) [inline src/test/app/claims/converters/claimantResponseConverter.ts]: why is there court determination if the claimant accepts defendants offer

File: src/test/app/claims/converters/claimantResponseConverter.ts [hunk 4]
```diff
@@ -140,157 +95,152 @@ describe('claimant response converter ', () => {
...
     it(' Accept defendant offer with settlement', () => {
       const draftClaimantResponse = createDraftClaimantResponseBaseForAcceptance(null,YesNoOption.YES)
       draftClaimantResponse.formaliseRepaymentPlan = new FormaliseRepaymentPlan(FormaliseRepaymentPlanOption.SIGN_SETTLEMENT_AGREEMENT)
-      draftClaimantResponse.decisionType = DecisionType.DEFENDANT
+      draftClaimantResponse.courtDetermination = courtDetermination
       expect(converter.convertToClaimantResponse(draftClaimantResponse)).to.deep.eq({
         'type': 'ACCEPTATION',
         'amountPaid': 10,
-        'formaliseOption': 'SETTLEMENT'
+        'formaliseOption': 'SETTLEMENT',
```
Comment (srineelesh) [inline src/test/app/claims/converters/claimantResponseConverter.ts]: why is there court determination if the claimant is accepting defendants offer

**Pull request merged**